jetbrains-core/it/software/aws/toolkits/jetbrains/services/cloudformation/annotations/CloudFormationLintAnnotatorTest.kt

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudformation.annotations
+
+import com.intellij.lang.xml.XMLLanguage
+import com.intellij.psi.PsiFileFactory
+import com.intellij.psi.xml.XmlFile
+import com.intellij.testFramework.runInEdtAndGet
+import org.assertj.core.api.Assertions.assertThat
+import org.jetbrains.yaml.YAMLLanguage
+import org.jetbrains.yaml.psi.YAMLFile
+import org.junit.Rule
+import org.junit.Test
+import software.aws.toolkits.jetbrains.utils.rules.JavaCodeInsightTestFixtureRule
+
+class CloudFormationLintAnnotatorTest {
+
+    @Rule
+    @JvmField
+    val projectRule = JavaCodeInsightTestFixtureRule()
+
+    var cloudFormationLintAnnotator: CloudFormationLintAnnotator = CloudFormationLintAnnotator()
+
+    @Test
+    fun canAnnotate() {
+        val psiFile = runInEdtAndGet {
+            PsiFileFactory.getInstance(projectRule.project).createFileFromText(
+                YAMLLanguage.INSTANCE, """
+Resources:
+  s3Bucket:
+    Type: undef
+        """.trimIndent()
+            ) as YAMLFile
+        }
+        val initialAnnotationResults = cloudFormationLintAnnotator.collectInformation(psiFile)
+        assertThat(initialAnnotationResults).isNotNull
+        val errors = cloudFormationLintAnnotator.doAnnotate(initialAnnotationResults)
+        assertThat(errors).isNotNull
+        assertThat(errors.size).isEqualTo(1)
+
+        // TODO test errorAnnotator.apply(), but I would need to be able to mock a static method
```

Comment:
What static method would you need to mock?

Reply:
` FileDocumentManager.getInstance()`

Reply:
per @abrooksv 's comment https://github.com/aws/aws-toolkit-jetbrains/pull/1715#discussion_r413301623 - we shouldn't need to mock the `FileDocumentManager` - that puts us on a treadmill.

Reply:
Ok I fixed this, though I'm not sure why code coverage doesn't show this as covered :(